mailbox/api/src/main/java/org/apache/james/mailbox/MailboxManager.java

```diff
@@ -238,7 +237,7 @@
      *            the context for this call, not null
      * @throws MailboxException
      */
-    Map<MailboxId, Collection<MessageUid>> search(MultimailboxesSearchQuery expression, MailboxSession session) throws MailboxException;
+    List<MessageId> search(MultimailboxesSearchQuery expression, MailboxSession session, long limit) throws MailboxException;
```

Comment:
1/ It's already implemented
2/ See above comment
3/ What we truely want here is messageId. I think it's a clear win to have GetMessageList querying only ES and not Cassandra.